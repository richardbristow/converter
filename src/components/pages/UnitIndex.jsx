import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/macro';

import StyledNavPage from '../shared/StyledNavPage';

const StyledUnitIndex = styled.div`
  padding-top: 20px;
  padding-bottom: 50px;
`;

const UnitIndex = ({ baseUnits }) => (
  <StyledNavPage>
    <h2>Complete Unit Index</h2>
    <StyledUnitIndex>
      {
        Object.keys(baseUnits).map((type) => {
          const unitObject = baseUnits[type];
          return (
            <Fragment key={type}>
              <h4>{unitObject.displayName}</h4>
              <p>{unitObject.units.map(unit => unit.displayName).join(', ')}</p>
            </Fragment>
          );
        })
      }
    </StyledUnitIndex>
  </StyledNavPage>
);

UnitIndex.propTypes = {
  baseUnits: PropTypes.objectOf(PropTypes.object).isRequired,
};

export default UnitIndex;
