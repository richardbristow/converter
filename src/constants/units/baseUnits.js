import addPrefixes from '../../utils/addPrefixes';

import length from './unitTypes/length';
import surfaceArea from './unitTypes/surfaceArea';
import volume from './unitTypes/volume';
import liquidVolume from './unitTypes/liquidVolume';
import angles from './unitTypes/angles';
import time from './unitTypes/time';
import frequency from './unitTypes/frequency';
import mass from './unitTypes/mass';
import electricCurrent from './unitTypes/electricCurrent';
import temperature from './unitTypes/temperature';
import force from './unitTypes/force';
import energy from './unitTypes/energy';
import power from './unitTypes/power';
import pressure from './unitTypes/pressure';
// import electricityAndMagnetism from './unitTypes/electricityAndMagnetism';
import datastorage from './unitTypes/dataStorage';

// import smallPrefixes from './prefixes/smallPrefixes';
// import largePrefixes from './prefixes/largePrefixes';
import metricPrefixes from './prefixes/metricPrefixes';
import datastoragePrefixes from './prefixes/dataStoragePrefixes';

const baseUnits = {
  length: addPrefixes(metricPrefixes, length),
  surfaceArea,
  volume,
  liquidVolume,
  angles,
  time: addPrefixes(metricPrefixes, time),
  frequency: addPrefixes(metricPrefixes, frequency),
  mass,
  electricCurrent: addPrefixes(metricPrefixes, electricCurrent),
  temperature,
  // amountOfSubstance,
  // luminousIntensity,
  // currency,
  force,
  energy,
  power,
  pressure,
  // electricityAndMagnetism,
  datastorage: addPrefixes(datastoragePrefixes, datastorage),
};

export default baseUnits;
