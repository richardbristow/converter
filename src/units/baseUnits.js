import length from './unitTypes/length';
import surfaceArea from './unitTypes/surfaceArea';
import time from './unitTypes/time';
import temperature from './unitTypes/temperature';

const baseUnits = {
  length,
  surfaceArea,
  // volume,
  // liquidVolume,
  // angles,
  time,
  // frequency,
  // mass,
  // electricCurrent,
  temperature,
  // amountOfSubstance,
  // luminousIntensity,
  // currency,
  // force,
  // energy,
  // power,
  // pressure,
  // electricityAndMagnetism,
  // binary,
};

export default baseUnits;
