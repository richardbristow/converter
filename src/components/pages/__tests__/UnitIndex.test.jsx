import React from 'react';
import { shallow } from 'enzyme';
import UnitIndex from '../UnitIndex';
import baseUnits from '../../../unitConstants/baseUnits';

describe('UnitIndex', () => {
  it('should render the Unit Index page', () => {
    const wrapper = shallow(<UnitIndex baseUnits={baseUnits} />);
    expect(wrapper).toMatchSnapshot();
  });
});
