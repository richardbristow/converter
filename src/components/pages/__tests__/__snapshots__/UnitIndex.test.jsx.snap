// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnitIndex should render the Unit Index page 1`] = `ShallowWrapper {}`;
